xds/src/main/java/io/grpc/xds/EnvoyServerProtoData.java

```diff
@@ -38,14 +38,60 @@
   private EnvoyServerProtoData() {
   }
 
-  public static final class DownstreamTlsContext {
+  public abstract static class BaseTlsContext {
+    protected final CommonTlsContext commonTlsContext;
+
+    public BaseTlsContext(CommonTlsContext commonTlsContext) {
```

Comment:
commonTlsContext is nullable?

Reply:
In a later PR I will translate CommonTlsContext to its internal version in which case default instance will be mapped to null and will be nullable but currently it will be non-null. I will mark it nullable. Practically speaking it is an error/bug for the xDS server to include UpstreamTlsContext with a default instance (as explained before).

Reply:
okay, i left the comment because there are few places checking nullness of commonTlsContext.